python/pyspark/sql/connect/client/core.py

```diff
@@ -296,7 +296,10 @@ def userAgent(self) -> str:
             or "_SPARK_CONNECT_PYTHON" when not specified.
             The returned value will be percent encoded.
         """
-        user_agent = self.params.get(ChannelBuilder.PARAM_USER_AGENT, "_SPARK_CONNECT_PYTHON")
+        user_agent = self.params.get(
+            ChannelBuilder.PARAM_USER_AGENT,
+            os.getenv("SPARK_CONNECT_USER_AGENT", "_SPARK_CONNECT_PYTHON"),
```

Comment:
Would like to call this `SPARK_CONNECT_USER_AGENT_EXTRA`. 

@grundprinzip - Is that acceptable to you?

Reply:
nvm. I found my answer elsewhere.